homework2/task1/for_text.py

```diff
@@ -30,28 +30,49 @@ def get_most_common_non_ascii_char(file_path: str) -> str:
 """
 import unicodedata
 from operator import itemgetter
-from typing import List
+from typing import Iterator, List
 
 
-def get_longest_diverse_words(file_path: str, encoding="utf8") -> List[str]:
-    dict_of_unique = {}
+def encoding_file(file_path: str, encoding="utf8") -> Iterator:
+    """
+    Generator that return lines from encoding file
+    :param file_path: path to file, str
+    :param encoding: mode to encode file, by default utf8 mode
+    :return: Iterator
+    """
     file_input = open(file_path, encoding=encoding)
     for line in file_input:
         line = line.encode().decode("unicode-escape")
+        yield line
```

Comment:
Сразу не заметил, но здесь `line.encode().decode("unicode-escape")` происходит кодирование обратной в байты и раскодирование с новой кодировкой. По-идее можно сразу читать из файла в новой кодировке чтобы не выполнять эту работу дважды

```suggestion
def encoding_file(file_path: str, encoding="unicode-escape") -> Iterator:
    """
    Generator that return lines from encoding file
    :param file_path: path to file, str
    :param encoding: mode to encode file, by default utf8 mode
    :return: Iterator
    """
    file_input = open(file_path, encoding=encoding)
    for line in file_input:
        yield line
```